### ISSUE TITLE
Sidebar category accessibility - child items not announced by screen reader

### ISSUE THREAD

**Issue by pkowaluk** [facebook/docusaurus#6913]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#reporting-new-issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [X] I have tried the `npm run clear` or `yarn clear` command.
- [X] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [X] I have tried creating a repro with https://new.docusaurus.io.
- [X] I have read the console error message carefully (if applicable).

### Description

I'm using VoiceOver on Mac (it's a screen reader) to check a11y. Category items in the sidebar are not announced as expandable items. They are announced as links and the user can "click" but they have no way of knowing there are subitems.

**I propose this solution**: it should be enough to add the aria-expanded attribute and it gets set to true or false (depending on the state). [aria-expanded - Accessibility | MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-expanded)

### Steps to reproduce

1. On Mac, turn on VoiceOver
2. In Chrome, go to https://docusaurus.io/docs
3. Using Tab on your keyboard, navigate to "Getting started"

### Expected behavior

VoiceOver should say "Link getting started, collapsed"

### Actual behavior

VoiceOver says "Link getting started"

### Your environment

- Public source code: n/a
- Public site URL: n/a
- Docusaurus version used: beta-17
- Environment name and version (e.g. Chrome 89, Node.js 16.4): Chrome 99
- Operating system and version (e.g. Ubuntu 20.04.2 LTS): Mac OS Monterey (12.2.1)

### Self-service

- [X] I'd be willing to fix this bug myself.

**Comment by Josh-Cena** [comment]
Thanks for the report, PRs are welcome!

Because categories are rendered lazily, I think `aria-expanded` would be the best solution.
